### PR TITLE
Math syntax support

### LINKED ISSUE #64: Math syntax support
Math support is very useful. We'll follow Obsidian who use mathjax (https://help.obsidian.md/How+to/Format+your+notes#Math).

Here's a guide to supporting this in markdown/mdx: https://mdxjs.com/guides/math/

Essence is to use two plugins https://github.com/remarkjs/remark-math/tree/main/packages/remark-math and https://github.com/remarkjs/remark-math/tree/main/packages/rehype-mathjax

## Acceptance

* [ ] Inline math like `$\sqrt{a^2 + b^2}$` works
* [ ] Block math works like the below
* [ ] Documentation on e.g. `/docs/math`

```
$$\begin{vmatrix}a & b\\
c & d
\end{vmatrix}=ad-bc$$
```

## Tasks

* [ ] Add the two plugins

### PR DESCRIPTION
**Pull request by khalilcodes**
## Add math syntax support with documentation
Resolves #64 
***
This PR adds support for using math syntax as in obsidian using the following remark and rehype plugins
- https://github.com/remarkjs/remark-math/tree/main/packages/remark-math
- https://github.com/remarkjs/remark-math/tree/main/packages/rehype-mathjax

### Tasks

- [x] add both remark and rehype plugins in contentlayer
- [x] ~~add custom Paragraph component to support some breaking styles with math syntax~~
- [x] add math inline styles in `styles/global.css`
- [x] create docs at [/docs/math](https://deploy-preview-77--spectacular-dragon-c1015c.netlify.app/docs/math)

### Notes

Block math only transforms when `$$` signs are above and below the syntax. for eg.
```markdown
$$
\begin{vmatrix}a & b\\
c & d
\end{vmatrix}=ad-bc
$$
```

### CODE REVIEW

File: site/content/docs/math.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+```
+
+will be tranformed as
+
+$$
+\begin{vmatrix}a & b\\
+c & d
+\end{vmatrix}=ad-bc
+$$
+
```
Comment (rufuspollock) [inline site/content/docs/math.md:40]: Trailing blank space. Maybe also add a short section for Developers e.g. `## Developers` that says that this uses mathjax or similar.

File: templates/default/components/MDX.js [hunk 2]
```diff
@@ -2,9 +2,18 @@ import Head from 'next/head'
 import DemoComponent from './Demo'
 import { Pre } from './Pre'
 
+const Paragraph = (props) => {
```
Comment (rufuspollock) [inline templates/default/components/MDX.js:5]: Why is this necessary? Looking at https://github.com/remarkjs/remark-math it seemed we could just use the remark plugins.

I'm guessing this is styling related. If so some comment here would have been recommended to explain this briefly.

Also if we used katex would this still be necessary?

And could we use traditional "css" for this? 

I ask as this seems a bit hacky and impinges on core code.

**Review by rufuspollock — COMMENTED** [review]
Have one substantive question - otherwise looks good.

File: templates/default/components/MDX.js (hunk 2 above)
Comment (khalilcodes) [reply templates/default/components/MDX.js]: Yes, this was style related. Moved to css and removed the component.

**Comment by rufuspollock** [comment]
@khalilcodes LGTM 👏 . Good work!
